### PR TITLE
MissionItem memset to 8 instead of 0

### PR DESCRIPTION
**Pull request by tilaktilak**
This memset seems to create an issue with GotoLocation on Ardupilot : https://github.com/ArduPilot/ardupilot/issues/12085

@DonLakeFlyer

### CODE REVIEW

File: src/MissionManager/MissionManager.cc [hunk 1]
```diff
@@ -46,7 +46,7 @@ void MissionManager::writeArduPilotGuidedMissionItem(const QGeoCoordinate& gotoC
     mavlink_message_t       messageOut;
     mavlink_mission_item_t  missionItem;
```
Comment (dagar) [inline src/MissionManager/MissionManager.cc]: You could also drop the memset entirely. 
```suggestion
    mavlink_mission_item_t  missionItem{};
```

Comment (DonLakeFlyer) [reply src/MissionManager/MissionManager.cc]: I think that will work. Not quite sure though. We've had problems with some of the newer c++ stuff working on all platforms.

**Comment by DonLakeFlyer** [comment]
@tilaktilak Can you create a pull for Stable_V3.5.1 as well?

**Comment by DonLakeFlyer** [comment]
Also since this is a fairly important bug fix it should be noted in changelog.md

**Pull request merged**